server/src/main/java/io/spine/server/event/EventBus.java

```diff
@@ -148,7 +154,28 @@ boolean hasDispatchers(EventClass eventClass) {
 
     @Override
     protected DeadMessageTap<EventEnvelope> getDeadMessageHandler() {
-        return DeadEventTap.INSTANCE;
+        return deadMessageHandler;
+    }
+
+    /**
+     * Creates a {@link DeadMessageTap} instance to handle a dead event by producing an 
+     * {@link UnsupportedEventException}, and saving the event to the Event Store.
+     *
+     * <p>An event gets into this tap if it does not pass the {@link io.spine.server.bus.DeadMessageFilter}.
+     * Although, this stops the Event Bus from handling the event, it still must be added to the Event Store.
+     */
+    private DeadMessageTap<EventEnvelope> newDeadEventTap() {
+        return new DeadMessageTap<EventEnvelope>() {
```

Comment:
Let's extract this as a inner (non-`static`) class. As I look at it right now, I don't like how complex this expression seems to be.